packages/core/lib/debug/printer.js

```diff
@@ -209,10 +209,33 @@ class DebugPrinter {
       this.config.logger.log(DebugUtils.formatStack(step.stack));
       this.config.logger.log("");
     }
-    this.config.logger.log(
-      DebugUtils.formatInstruction(traceIndex + 1, totalSteps, instruction)
-    );
-    this.config.logger.log(DebugUtils.formatPC(step.pc));
+
+    // printout instructions
+    const previousInstructions = 3;
+    const upcomingInstructions = 3;
+    const currentInstructionIndex = instructions.findIndex(instruction => instruction.pc === step.pc);
```

Comment:
You don't need to iterate over `instructions` to find the current index!  It's already stored in the `index` field in `solidity.current.instruction`.  This is why I pointed that out to you earlier that you want to use both selectors; you can do this with a constant-time access, you don't need a linear-time search.

Reply:
Thanks for pointing this out.  Change to use the index field of solidity.current.instruction.

Reply:
Btw, note that if you did need to look it up by `pc` rather than by `index`, you could use the `solidity.current.instructionAtProgramCounter` selector.  But obviously there isn't actually a need for that here. :)